### DISCUSSION TITLE
Cannot migrate to 6.4.10

### DISCUSSION THREAD

**Discussion by jonasthern** [fluentassertions/fluentassertions#2149] [General]
### Description

Migrating a unit test project from 5.10.3 to 6.10.0

All changes were applied but a missing one:

```
`Error	CS0121	The call is ambiguous between the following methods or properties: 'DataRowAssertionExtensions.Should<TDataRow>(TDataRow)' and 'DataSetAssertionExtensions.Should<TDataSet>(TDataSet)'
```

Also running this on our pipeline with vsnet 2017.

### Reproduction Steps

Any instruction with Should gives error when compiling:
```csharp
List<CompareResult> compareResults = compareTreeService.Compare(sourceNode, targetNode).ToList();

compareResults.Should().HaveCount(nrOfChildren + 1);
```

### Expected behavior

No compilation errors.

### Actual behavior

All instructions gives the error:
```
Error	CS0121	The call is ambiguous between the following methods or properties: DataRowAssertionExtensions.Should<TDataRow>(TDataRow)' and 'DataSetAssertionExtensions.Should<TDataSet>(TDataSet)
```

### Known Workarounds

- already tried langversion but does not apply to us

### Configuration

- net project is 4.7.2
- unit tests
- using packages.config
- building in vsnet2017

**Comment by jnyrup** [comment]
This is fixed in C# 7.3 with [Improved overload candidates](https://github.com/dotnet/csharplang/blob/main/proposals/csharp-7.3/improved-overload-candidates.md).

In a previous version of this issue, you specified that for compatibility reasons you cannot upgrade your project from C# 7.1.

A workaround for C# 7.1 is to use "static using" to help overload resolution to disambiguate the `Should`.

```cs
using Microsoft.VisualStudio.TestTools.UnitTesting;
using System.Collections.Generic;

using static FluentAssertions.AssertionExtensions;

namespace TestProject6
{
    [TestClass]
    public class UnitTest1
    {
        [TestMethod]
        public void TestMethod1()
        {
            var compareResults = new List<int>() { 42 };
            compareResults.Should().HaveCount(1);
        }
    }
}
```

**Reply by jonasthern** [reply]
Thank you. I had the chance to upgrade to 7.3 and it worked.

**Reply by jnyrup** [reply]
Glad to hear it worked out. 
I learned something new as well.
